test/test.js

```diff
@@ -599,54 +278,78 @@ describe('avltree-js tests', function() {
 			tree.add(11);
 			tree.add(2);
 			tree.removeByReference(three);
-			assert.strictEqual(tree.root.object, 5);
-			assert.strictEqual(tree.root.height, 3);
-
-			assert.strictEqual(tree.root.left.object, 2);
-			assert.strictEqual(tree.root.left.height, 2);
-			assert.strictEqual(tree.root.right.object, 10);
-			assert.strictEqual(tree.root.right.height, 2);
-
-			assert.strictEqual(tree.root.left.left.object, 1);
-			assert.strictEqual(tree.root.left.left.height, 1);
-			assert.strictEqual(tree.root.left.right.object, 4);
-			assert.strictEqual(tree.root.left.right.height, 1);
-			assert.strictEqual(tree.root.right.right.object, 11);
-			assert.strictEqual(tree.root.right.right.height, 1);
+
+			assertTree(tree, 6);
 		});
 		it('should not crash if the tree is empty', function () {
 			tree.removeByReference(5);
+
+			assertTree(tree, 0);
 		});
 		it('should not crash if the object is not in the tree', function () {
 			tree.add(4);
 			tree.removeByReference(5);
+
+			assertTree(tree, 1);
 		});
 		it('should clear, nodes containers should be nulled out', function () {
 			var a = tree.add(1);
 			var b = tree.add(2);
 			var c = tree.add(3);
 			tree.clear();
+
+			assertTree(tree, 0);
+
 			assert.strictEqual(a.container, null);
 			assert.strictEqual(b.container, null);
 			assert.strictEqual(c.container, null);
 
-			assert.strictEqual(a.left, null);
-			assert.strictEqual(a.right, null);
-			assert.strictEqual(a.parent, null);
-			assert.strictEqual(a.height, 1);
+			assert.strictEqual(tree.root, null);
+			assert.strictEqual(tree.length, 0);
+		});
 
-			assert.strictEqual(b.left, null);
-			assert.strictEqual(b.right, null);
-			assert.strictEqual(b.parent, null);
-			assert.strictEqual(b.height, 1);
+		it('should become empty, large addition and removal', function () {
+			var nbElements = 5000;
+			var references = [];
+			for (var i = 0; i < nbElements; i += 1) {
+				references[i] = tree.add(Math.round(Math.random() * 100));
+			}
 
-			assert.strictEqual(c.left, null);
-			assert.strictEqual(c.right, null);
-			assert.strictEqual(c.parent, null);
-			assert.strictEqual(c.height, 1);
+			assertTree(tree, nbElements);
 
-			assert.strictEqual(tree.root, null);
-			assert.strictEqual(tree.length, 0);
+			var half = Math.round(nbElements / 2);
+
+			for (var i = half; i < nbElements; i += 1) {
+				tree.removeByReference(references[i]);
+			}
+
+			assertTree(tree, half);
+
+			for (var i = 0; i < half; i += 1) {
+				tree.removeByReference(references[i]);
+			}
+
+			assertTree(tree, 0);
+		});
+
+		it('should remain stable, large addition and removal', function () {
```

Comment:
takes a long time to run on my pc. mocha says 322ms.
Shouldn't do long running things in unit tests.  Maybe a separate test file for this one.  Should also be a good place to put these benchmarks you were speaking of.